repl/src/main/scala/com/cloudera/livy/repl/Session.scala

```diff
@@ -168,4 +190,24 @@ class Session(interpreter: Interpreter, stateChangedCallback: SessionState => Un
 
     compact(render(resultInJson))
   }
+
+  private def executeWithJobGroup(executionCount: Int, code: String): String = {
+    val cmd = Kind(interpreter.kind) match {
+      case Spark() | PySpark() | PySpark3() =>
+        s"""sc.setJobGroup($executionCount, "Job group for statement $executionCount")"""
```

Comment:
miss quotation mark around `$executionCount`, this make the spark job not canceled